packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -0,0 +1,428 @@
+      size: event.size,
+      radiusMajor: event.radiusMajor,
+      radiusMinor: event.radiusMinor,
+      radiusMin: event.radiusMin,
+      radiusMax: event.radiusMax,
+      orientation: event.orientation,
+      tilt: event.tilt,
+      synthesized: event.synthesized,
+      embedderId: event.embedderId,
+    );
+  }
+
+  PointerEvent _toMoveEvent(PointerEvent event, Offset position, Offset delta,
+      int pointerIdentifier, Duration timeStamp) {
+    return PointerMoveEvent(
+      timeStamp: timeStamp,
+      pointer: pointerIdentifier,
+      kind: event.kind,
+      device: event.device,
+      position: position,
+      delta: delta,
+      buttons: event.buttons,
+      obscured: event.obscured,
+      pressure: event.pressure,
+      pressureMin: event.pressureMin,
+      pressureMax: event.pressureMax,
+      distanceMax: event.distanceMax,
+      size: event.size,
+      radiusMajor: event.radiusMajor,
+      radiusMinor: event.radiusMinor,
+      radiusMin: event.radiusMin,
+      radiusMax: event.radiusMax,
+      orientation: event.orientation,
+      tilt: event.tilt,
+      platformData: event.platformData,
+      synthesized: event.synthesized,
+      embedderId: event.embedderId,
+    );
+  }
+
+  Offset _positionAt(Duration sampleTime) {
+    // Use `next` position by default.
+    double x = _next?.position.dx ?? 0.0;
+    double y = _next?.position.dy ?? 0.0;
+
+    final Duration nextTimeStamp = _next?.timeStamp ?? const Duration();
+    final Duration lastTimeStamp = _last?.timeStamp ?? const Duration();
+
+    // Resample if `next` time stamp is past `sampleTime`.
+    if (nextTimeStamp > sampleTime && nextTimeStamp > lastTimeStamp) {
+      final double interval =
+          (nextTimeStamp - lastTimeStamp).inMicroseconds.toDouble();
+      final double scalar =
+          (sampleTime - lastTimeStamp).inMicroseconds.toDouble() / interval;
+      final double lastX = _last?.position.dx ?? 0.0;
+      final double lastY = _last?.position.dy ?? 0.0;
+      x = lastX + (x - lastX) * scalar;
+      y = lastY + (y - lastY) * scalar;
+    }
+
+    return Offset(x, y);
+  }
+
+  void _processPointerEvents(Duration sampleTime) {
+    final Iterator<PointerEvent> it = _queuedEvents.iterator;
+    while (it.moveNext()) {
+      final PointerEvent event = it.current;
+
+      // Update both `last` and `next` pointer event if time stamp is older
+      // or equal to `sampleTime`.
+      if (event.timeStamp <= sampleTime || _last == null) {
+        _last = event;
+        _next = event;
+        continue;
+      }
+
+      // Update only `next` pointer event if time stamp is more recent than
+      // `sampleTime` and next event is not already more recent.
+      final Duration nextTimeStamp = _next?.timeStamp ?? const Duration();
+      if (nextTimeStamp < sampleTime) {
+        _next = event;
+        break;
+      }
+    }
+  }
+
+  List<PointerEvent> _dequeueAndSamplePointerEventsUntil(Duration sampleTime) {
```

Comment:
Nit: maybe it's a little clearer to name this as `_dequeueAndSampleNonHoverOrMovePointerEventsUntil`, or maybe keep the current name and add the following lines to the end of it
```
    // Add resampled pointer location event if tracked.
    if (_isTracked) {
      events.addAll(_samplePointerPositions(sampleTime));
    }
```

Reply:
Done.